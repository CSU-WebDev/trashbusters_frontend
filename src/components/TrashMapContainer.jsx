import React from 'react'
import TrashMap from './TrashMap';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export default function TrashMapContainer() {
    return (
        <>
            <TrashMap />
        </>
    )
}