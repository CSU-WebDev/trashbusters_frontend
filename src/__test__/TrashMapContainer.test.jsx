import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TrashMapContainer from '../components/TrashMapContainer';

describe('TrashMapContainer component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<TrashMapContainer />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Trash Map heading', () => {
    render(<TrashMapContainer />);
    expect(screen.getByText('Trash Map')).toBeInTheDocument();
  });

  test('renders TrashMap component', () => {
    render(<TrashMapContainer />);
    const trashMapContainer = screen.getByTestId('trashmap-container');
    expect(trashMapContainer).toBeInTheDocument();
  });
});
