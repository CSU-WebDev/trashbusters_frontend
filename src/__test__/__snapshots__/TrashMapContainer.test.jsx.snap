// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrashMapContainer component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box container"
    data-testid="trashmap-container"
    id="trashmap-container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="text-center"
        >
          Trash Map
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
        id="trashmap-container-col"
      >
        <div
          id="trashmap"
        >
          <div />
          <div>
            Loading...
          </div>
          <div
            id="spinnerDiv"
          >
            <div
              class="spinner-border text-dark"
              id="spinner"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
